### DISCUSSION TITLE
New beta4 MQTT switches

### DISCUSSION THREAD

**Discussion by S-Przybylski** [blakeblackshear/frigate#531] [General]
Dear @blakeblackshear and @paularmstrong thanks for the current "beta4" version of frigate. 
Today i build my docker for armv7 and manually updated my HACS custom_component. Its works like a charm!
I have two quesitons about thew switches for enabling/disabling snapshots, clips, ...
- What is the logic for the initial value? Whats happend by restarting HA or Frigate or changing the config.yml? 
  While i started the new version first time, some of my cameras had different states then described in the config.yml...
- Do I need to attach all roles to the camera stream to get the switches run?

By the way: 
- I had to change the dockerfile.base: I changed the way of installing python and add setproctitle:
````
RUN apt-get -qq update \
    && apt-get upgrade -y \
    && apt-get -qq install --no-install-recommends -y \
    gnupg wget unzip tzdata nginx libnginx-mod-rtmp \
    && apt-get -qq install -y python3-pip \
    && pip3 install setproctitle \ 
    && pip3 install -U /wheels/*.whl \
  ....
````
- Currently the Wheels image for armv7 isn't updated to the new naming convention...

**Comment by blakeblackshear** [comment]
> What is the logic for the initial value? Whats happend by restarting HA or Frigate or changing the config.yml?
While i started the new version first time, some of my cameras had different states then described in the config.yml

The switches publish to the MQTT topic with the retain flag, so MQTT is retaining the current switch status. Both HA and Frigate should sync up with MQTT on restart.

> Do I need to attach all roles to the camera stream to get the switches run?

I am not sure what you are asking here.

> By the way

I am fixing the docker build issues before publishing the release candidate.

**Comment by S-Przybylski** [comment]
Sorry i my question was unclear!
To use the binary sensors: Is it required to add the necessary roles (i believe detect and clips) to the camera stream as a requirement like before? 

Perhaps it could then be useful to define one of the streams as the default. If not further defined (the role assignment), this camera stream is used for providing all required roles... Only if I add an additional stream, i could add clips or recording to that as an alternative configuration .... Of course rtmp is a special case ( in my setup i don't use the feature, because i have huge delays; i prefer the mjpeg)

**Comment by blakeblackshear** [comment]
You must have the roles assigned for the switches to have any effect. These can't be dynamically assigned because the ffmpeg output parameters are defined by the roles. These switches do not stop and start ffmpeg processes, they just drive frigate's behavior.

**Reply by S-Przybylski** [reply]
ok thanks
I have an additional question regarding the hass-add on: Is it possible to change the default icon of the switches?
I found some more or less useful ones:
clips: mdi:filmstrip-box-multiple
snapshot: mdi:image-multiple
detect: hass:motion-sensor

**Reply by blakeblackshear** [reply]
updated. thanks

**Comment by S-Przybylski** [comment]
And sorry for the missunderstanding I always meant switches instead of binary_sensors!
